priiloader/source/main.cpp

```diff
@@ -3519,8 +3517,18 @@ int main(int argc, char **argv)
 	flags = GetStateFlags();
 	gprintf("Bootstate %u detected. DiscState %u ,ReturnTo %u & Flags %u & checksum %u\n",flags.type,flags.discstate,flags.returnto,flags.flags,flags.checksum);
 	s8 magicWord = CheckMagicWords();
+	
+	// before anything else, poll input devices such as keyboards to see if we should stop autoboot
+	r = Input_Init();
+	gprintf("Input_Init():%d\n", r );
+	
+	WPAD_SetPowerButtonCallback(HandleWiiMoteEvent);
+	
+	usleep(500000); // Give things enough time to settle down
+	Input_ScanPads();
+	
 	//Check reset button state
-	if( (((*(vu32*)0xCC003000)>>16)&1) == 1 && magicWord == 0) //if( ((*(vu32*)0xCC003000)>>16)&1 && !CheckMagicWords())
+	if(((Input_ButtonsDown() & INPUT_BUTTON_START) == 0) && (((*(vu32*)0xCC003000)>>16)&1) == 1 && magicWord == 0) //if( ((*(vu32*)0xCC003000)>>16)&1 && !CheckMagicWords())

```

Comment:
just commenting out loud that , though start is not bad of a choice, i was thinking if adding a new input button wasn't a bad idea for this. as is, this will allow a GC start or wiimote + to open the menu

Reply:
I'll add it in as ESC if you like, although that would make impossible to hit on a gamecube controller. I choose to just go with a button that was common to all of them because of that.

Reply:
a ye ok, if it was your intention to add the input as a button for every input device then its also good.

personally, i'd map it to a different value though. something like INPUT_BUTTON_HOME to indicate its purpose better.
however, the gamecube annoys me a bit. 
for the wiimote it could be home & keyboard gets esc (since 80% or 60% keyboards don't always have home and escape just sounds better for its purpose).
but the gamecube has all buttons mapped i believe so it could only fit as start...

Reply:
I think I'll leave it as it is unless you have any strong want to change it. I want to keep it available on all input devices and not have it be mapped to a weird keyboard key that might not exist on someones keyboard.